compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -325,8 +325,8 @@ pub fn super_relate_tys<R: TypeRelation<'tcx>>(
 ) -> RelateResult<'tcx, Ty<'tcx>> {
     let tcx = relation.tcx();
     debug!("super_relate_tys: a={:?} b={:?}", a, b);
-    match (&a.kind(), &b.kind()) {
-        (&ty::Infer(_), _) | (_, &ty::Infer(_)) => {
+    match (a.kind(), b.kind()) {
```

Comment:
I hope this was the main reason for the regression. Relate is quite hot and `&a.kind()` has quite bad codegen